core/main/src/broker/thunder_broker.rs

```diff
@@ -207,7 +208,17 @@ impl ThunderBroker {
         let broker_for_cleanup = thunder_broker.clone();
         let broker_for_reconnect = thunder_broker.clone();
         tokio::spawn(async move {
-            let (ws_tx, mut ws_rx) = BrokerUtils::get_ws_broker(&endpoint.get_url(), None).await;
+            let resp = WebSocketUtils::get_ws_stream(&endpoint.get_url(), None).await;
+            if resp.is_err() {
+                error!("FATAL error Thunder URL badly configured.");
+                // This stops the Server
+                let reconnect_request = request.clone();
+                if request.reconnector.send(reconnect_request).await.is_err() {
+                    error!("Error trying to stop server");
+                }
+                return;
+            }
+            let (ws_tx, mut ws_rx) = resp.unwrap();
```

Comment:
Unsafe unwrap(). 

```
let (ws_tx, mut ws_rx) = match resp {
    Ok(stream) => stream,
    Err(_) => <--- move the error handling code to here.  
};
```

Move the error handling code from `resp.is_err()` to the Err(__) matching arm